cloudevents/working-drafts/xml-format.md

```diff
@@ -0,0 +1,243 @@
+
+``` xml
+<DataXml>
+        <myData>
+            ....
+        </myData>  
+</DataXml>
+```
+
+## 4. Envelope
+
+Each CloudEvent is wholly represented as an XML element, the root xml
+document element is `<CloudEvent>`. This element carries the `speversion`
+as an XML attribute value.
+
+Such a representation MUST use the media type `application/cloudevents+xml`.
+
+The enveloping element contains:
+
+* A set of context attributre elements.
+* Optionally: Either `<Data>` or `<DataXml>`
+
+eg (Namespace definitions omitted for brevity):
+
+``` xml
+<CloudEvent specversion="1.0">
+    <time>2020-03-19T12:54:00-07:00</time>
+    <datacontenttype>text/plain</datacontenttype>
+    <id>000-1111-2222</id>
+    <source>urn:uuid:123e4567-e89b-12d3-a456-426614174000</source>
+    <type>SOME.EVENT.TYPE</type>
+    <myboolean xsi:type="xs:boolean">false</myboolean>
+    <Data xsi:type="xs_string">Now is the winter of our discount tents...</Data
+</CloudEvent>
+```
+
+## 5. XML Batch Format
+
+In the _XML Batch Format_ several CloudEvents are batched into a single XML
+document. The document comprises a list of elements in the XML Format.
+
+Although the _XML Batch Format_ builds on top of the _XML Format_, it is
+considered as a separate format: a valid implementation of the _XML Format_
+doesn't need to support it. The _XML Batch Format_ MUST NOT be used when only
+support for the _XML Format_ is indicated.
+
+An XML Batch of CloudEvents MUST use the media type
+`application/cloudevents-batch+xml`.
+
+Example:
+
+``` xml
+<CloudEventBatch>
+    <CloudEvent specverison="1.0">
+       ....
+    </CloudEvent>
+    <CloudEvent specversion="1.0">
+       ....
+    </CloudEvent>
+    ....
+</CloudEventBatch>
+```
+
+An example of an empty batch of CloudEvents (typically used in a response, but also valid in a request):
+
+```xml
+<CloudEventBatch />
+```
+
+## 6. Examples
+
+### 6.1 CloudEvent with PNG image data
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<CloudEvent xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xs="http://www.w3.org/2001/XMLSchema" specversion="1.0" >
+    <time>2020-03-19T12:54:00-07:00</time>
+    <datacontenttype>image/png</datacontenttype>
+    <id>000-1111-2222</id>
+    <source>urn:uuid:123e4567-e89b-12d3-a456-426614174000</source>
+    <type>SOME.EVENT.TYPE</type>
+    <Data xsi:type="xs:base64Binary">... Base64 encoded data...</Data>
+</CloudEvent>
+```
+
+### 6.2 CloudEvent with JSON event data
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<CloudEvent xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xs="http://www.w3.org/2001/XMLSchema" specversion="1.0" >
+    <time>2020-03-19T12:54:00-07:00</time>
+    <datacontenttype>application/json</datacontenttype>
+    <id>000-1111-2222</id>
+    <source>urn:uuid:123e4567-e89b-12d3-a456-426614174000</source>
+    <type>SOME.EVENT.TYPE</type>
+    <Data xsi:type="xs:string">{ "salutation" : "Good Morning", "text" : "hello world" }</Data>
```

Comment:
what happens if the JSON has a `<` in it? Is there any escaping needed?

Reply:
Yes ... content does need to be appropriately escaped to ensure the XML document is well formed.

Reply:
Can we remove the spaces before the colons please? in other words:

```xml
<Data xsi:type="xs:string">{ "salutation": "Good Morning", "text": "hello world" }</Data>
```